driver/modern_bpf/helpers/store/auxmap_store_params.h

```diff
@@ -153,7 +145,50 @@ static __always_inline void auxmap__submit_event(struct auxiliary_map *auxmap, v
 	{
 		counter->n_drops_buffer++;
 		compute_event_types_stats(auxmap->event_type, counter);
+		return;
+	}
+	return;
+}
+
+/**
+ * @brief Try to copy the entire event from the auxiliary map to bpf ringbuf.
+ * If the event is correctly copied in the ringbuf we increment the number
+ * of events sent to userspace, otherwise we increment the dropped events.
+ *
+ * @param auxmap pointer to the auxmap in which we have already written the entire event.
+ * @returns 0 if we got the ringbuffer correctly.
+ */
+static __always_inline int auxmap__try_submit_event(struct auxiliary_map *auxmap)
+{
+	struct ringbuf_map *rb = maps__get_ringbuf_map();
+	if(!rb)
+	{
+		return 1;
 	}
+
+	auxmap__submit_event_base(auxmap, rb);
+	return 0;
+}
+
+/**
+ * @brief Copy the entire event from the auxiliary map to bpf ringbuf.
+ * If the event is correctly copied in the ringbuf we increment the number
+ * of events sent to userspace, otherwise we increment the dropped events.
+ *
+ * @param auxmap pointer to the auxmap in which we have already written the entire event.
+ * @param ctx BPF prog context
+ */
+static __always_inline void auxmap__submit_event(struct auxiliary_map *auxmap, void* ctx)
```

Comment:
Just as a note on why this function had to be split into a 3, a base, a try and this one:

The new implementation for sendmmsg/recvmmsg using bpf_loop require the function passed as a callback parameter to be just that, a callback, meaning clang is not able to inline it like it does with all other functions. This in turn means that these new tracepoints produce bpf-to-bpf calls which are only allowed to do tail calls in JITed programs. That's all well and good, it works perfectly on x86, but it turns out it was implemented on arm, s390x and ppc64le only a couple years ago, which means it breaks a bunch of our integration tests with a verifier error. In order to fix this, I split the `auxmap__submit_event` so I could create a separate version that does not do the tail call, thus having our bpf-to-bpf call programs not do any tail calling themselves.

For additional context, the tail call is responsible for triggering a restart when a hotplug CPU is brought online, this is obviously crucial and should be done ASAP, not doing that tail call in these syscalls means that we will delay restart until another syscall is triggered. Given that restarting collector takes a non trivial amount of time, I think it's fine that we miss a couple syscalls, since the restart will cause us to drop quite a bit more than those.